### PR TITLE
Light refactoring to move some logic from the view layer into the store

### PR DESCRIPTION
**Pull request by graynorton**
This PR moves some logic from the view layer into the store.

The goal is to make the separation between the data layer and the view layer a bit cleaner and make the view components easier to grok at a glance; for the most part, they only implement LitElement lifecycle methods and `stateChanged()`.

### CODE REVIEW

File: src/reducers/shop.js [hunk 1]
```diff
@@ -136,3 +137,43 @@ const quantityById = (state = INITIAL_CART.quantityById, action) => {
...
 
 export default shop;
+
+const cartSelector = state => state.shop.cart;
+const productsSelector = state => state.shop.products;
+
+export const cartItemsSelector = createSelector(
+  cartSelector,
+  productsSelector,
+  (cart, products) => {
```
Comment (notwaldorf) [inline src/reducers/shop.js]: In the interest of this being a learning template, what do you think about excessively documenting these? I.e. a generic comment about what a selector is, why these 5 are here, what each does maybe?

File: src/components/my-view3.js [hunk 2]
```diff
@@ -23,13 +23,13 @@ import { store } from '../store.js';
 import { checkout } from '../actions/shop.js';
 
 // We are lazy loading its reducer.
-import shop from '../reducers/shop.js';
+import shop, { cartQuantitySelector } from '../reducers/shop.js';
```
Comment (notwaldorf) [inline src/components/my-view3.js]: omg what is this syntax. should it be `{shop, cartQuantitySelector}` instead? My brain is imploding

**Comment by notwaldorf** [comment]
I like this and the reselects quite a bit!  Just a minor comment left

File: src/reducers/shop.js (hunk 1 above)
Comment (graynorton) [reply src/reducers/shop.js]: Absolutely! I'll add some.

File: src/components/my-view3.js (hunk 2 above)
Comment (graynorton) [reply src/components/my-view3.js]: This is what it looks like when you import both a module's default export and one of its named exports.

Previously the shop reducer module only exported one thing (the reducer itself), as a default export. When I added named exports for the selectors, I could have removed the default export and instead exported the reducer as a named export, but I didn't do that.

The upside of removing the default and going with only named exports would be less confusing import syntax (per your example). The downside is that this module would be exporting its "main" thing in a different way from the other reducer modules in the template...unless we changed all of them to use named exports as well.

Thoughts? (cc @frankiefu, @keanulee)

Comment (notwaldorf) [reply src/components/my-view3.js]: imo i'd do anything to avoid this syntax lol

Comment (frankiefu) [reply src/components/my-view3.js]: I don't have a strong opinion on this. But I think this kind of syntax the users need to deal with and learn anyway if you import some other libraries and that libraries may have default and named in the same file.

File: src/reducers/shop.js (hunk 1 above)
Comment (graynorton) [reply src/reducers/shop.js]: Done!

**Pull request merged**